maps_dashboards/public/components/map_top_nav/get_top_nav_config.tsx

```diff
@@ -29,19 +40,36 @@ export const getTopNavConfig = (services: MapServices) => {
         defaultMessage: `Save`,
       }),
       run: (_anchorElement) => {
-        const onModalSave = async (onSaveProps: OnSaveProps) => {
-          const savedMap = await savedObjectsClient.create('map', {
-            title: onSaveProps.newTitle,
-            description: onSaveProps.newDescription,
-            // TODO: Integrate other attributes to saved object
-          });
-          const id = savedMap.id;
+        const onModalSave = async ({ newTitle, newDescription }: OnSaveProps) => {
+          let newlySavedMap;
+          if (mapIdFromUrl) {
+            // edit existing map
+            newlySavedMap = await savedObjectsClient.update('map', mapIdFromUrl, {
```

Comment:
instead of using the saved object client directly, consider using the `SavedObjectLoader`. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.4/src/plugins/saved_objects/public/saved_object/saved_object_loader.ts#L36-L37

This lets you use a single saved object loader instace across the app that always operates on just the `map` saved object type. It is also Typesafe and you can avoid such if else conditions.

https://github.com/ashwin-pc/OpenSearch-Dashboards/blob/061e5bd8706c3a6a1aa089e6490c56397aa6b60a/src/plugins/vis_builder/public/saved_visualizations/_saved_vis.ts#L14

This can also e done in a future iteration but its one thing that will make working with saved objects a lot more straightforward, especially when it comes to future integrations with the SaveObjectManagement plugin, etc.

Reply:
Yeah that confused me a little. We dont have the best guidance on saved objects right now so thats on us to improve. The loader makes sense in your case since you always interact with a single saved object type to load and save all map data, a use case best suited for the loader. Backlog is good for now.